tools/building.py

```diff
@@ -509,14 +510,33 @@ def add_to_path(dirname):
   return closure_cmd, env
 
 
+def version_split(v):
+  v = str(v)
+  assert len(v) == 6
+  rev = int(v[4:6])
+  minor = int(v[2:4])
+  major = int(v[0:2])
+  return f'{major}.{minor}.{rev}'
```

Comment:
```suggestion
def version_split(v):
    v = str(v)
    assert len(v) == 6
    
    pattern = re.compile(r'(?P<major>\d{2})(?P<minor>\d{2})(?P<rev>\d{2})')
    match = pattern.match(v)
    
    if match:
        major = int(match.group('major'))
        minor = int(match.group('minor'))
        rev = int(match.group('rev'))
        return f'{major}.{minor}.{rev}'
    else:
        raise ValueError("Invalid version format")
```

Remember to `import re` at the beginning.

Reply:
I think verifying that the `MIN_XX_VERSION` settings are sane belongs elsewhere in the code and we can just assert that they are here.